nominatim/clicmd/setup.py

```diff
@@ -81,36 +85,46 @@ def run(self, args: NominatimArgs) -> int: # pylint: disable=too-many-statements
             if not files:
                 raise UsageError("No input files (use --osm-file).")
 
-            LOG.warning('Creating database')
-            database_import.setup_database_skeleton(args.config.get_libpq_dsn(),
-                                                    rouser=args.config.DATABASE_WEBUSER)
-
-            LOG.warning('Setting up country tables')
-            country_info.setup_country_tables(args.config.get_libpq_dsn(),
-                                              args.config.lib_dir.data,
-                                              args.no_partitions)
-
-            LOG.warning('Importing OSM data file')
-            database_import.import_osm_data(files,
-                                            args.osm2pgsql_options(0, 1),
-                                            drop=args.no_updates,
-                                            ignore_errors=args.ignore_errors)
-
-            LOG.warning('Importing wikipedia importance data')
-            data_path = Path(args.config.WIKIPEDIA_DATA_PATH or args.project_dir)
-            if refresh.import_wikipedia_articles(args.config.get_libpq_dsn(),
-                                                 data_path) > 0:
-                LOG.error('Wikipedia importance dump file not found. '
-                          'Calculating importance values of locations will not '
-                          'use Wikipedia importance data.')
-
-            LOG.warning('Importing secondary importance raster data')
-            if refresh.import_secondary_importance(args.config.get_libpq_dsn(),
-                                                   args.project_dir) != 0:
-                LOG.error('Secondary importance file not imported. '
-                          'Falling back to default ranking.')
-
-            self._setup_tables(args.config, args.reverse_only)
+            if args.no_superuser and args.prepare_database:
```

Comment:
The logic here is quite confusing, maybe renaming the options to `--only-prepare-database` and `--only-import-data` would help? Alternatively, split the database preparation completely out of `import` so you first do `nominatim prepare-db`, then `nominatim import --osm-file ...`. This would also more easily allow passing different postgres connection parameters to the prepare step, doing only that part as the superuser is the whole point after all.

Reply:
I agree, I'll start with implementing the different naming of the flags in favor of keeping backwards compatibility.